enterprise_gateway/services/processproxies/processproxy.py

```diff
@@ -369,10 +371,21 @@ def _create_ssh_tunnel(self, port_name, local_port, remote_port, remote_ip, ssh_
         self.log.debug("Creating SSH tunnel for '{}': 127.0.0.1:'{}' to '{}':'{}'"
                        .format(port_name, local_port, remote_ip, remote_port))
         try:
-            tunnel.ssh_tunnel(local_port, remote_port, ssh_server, remote_ip, ssh_key, timeout=maxint)
+            process = self._spawn_ssh_tunnel(port_name, local_port, remote_port, remote_ip, ssh_server, ssh_key=None)
+            self.tunnel_processes.append(process)
         except Exception as e:
             raise RuntimeError("Could not open SSH tunnel for port {}. Exception: '{}'".format(port_name, e))
 
+    def _spawn_ssh_tunnel(self, port_name, local_port, remote_port, remote_ip, ssh_server, ssh_key=None):
+        """ This method spawns a child process to create a SSH tunnel and returns the spawned process.
+        """
+        if sys.platform == 'win32':
+            return tunnel.paramiko_tunnel(local_port, remote_port, ssh_server, remote_ip, ssh_key, timeout=maxint)
+        else:
+            cmd = "%s -S none -L 127.0.0.1:%i:%s:%i %s sleep %i" % (
```

Comment:
It might be good to add a comment as to why the openssh version is implemented here without the `-f` option.  That prevents the `ssh` process from getting orphaned - which invalidates any handles to that process.  In addition, the zmq implementation returns the `pid` for the process rather than the process handle itself (unlike paramiko's implementation), so this has the added benefit of consistent return values from both implementations - where we can then terminate the tunnel processes during shutdown.

Reply:
Done